internal/service/quicksight/data_set.go

```diff
@@ -0,0 +1,2726 @@
+												},
+												"tags": {
+													Type:     schema.TypeList,
+													Required: true,
+													MinItems: 1,
+													MaxItems: 16,
+													Elem: &schema.Resource{
+														Schema: map[string]*schema.Schema{
+															"column_description": {
+																Type:     schema.TypeList,
+																Computed: true,
+																Optional: true,
+																MaxItems: 1,
+																Elem: &schema.Resource{
+																	Schema: map[string]*schema.Schema{
+																		"text": {
+																			Type:         schema.TypeString,
+																			Computed:     true,
+																			Optional:     true,
+																			ValidateFunc: validation.StringLenBetween(0, 500),
+																		},
+																	},
+																},
+															},
+															"column_geographic_role": {
+																Type:         schema.TypeString,
+																Computed:     true,
+																Optional:     true,
+																ValidateFunc: validation.StringInSlice(quicksight.GeoSpatialDataRole_Values(), false),
+															},
+														},
+													},
+												},
+											},
+										},
+									},
+									"untag_column_operation": {
+										Type:     schema.TypeList,
+										Computed: true,
+										Optional: true,
+										MaxItems: 1,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"column_name": {
+													Type:         schema.TypeString,
+													Required:     true,
+													ValidateFunc: validation.StringLenBetween(1, 128),
+												},
+												"tag_names": {
+													Type:     schema.TypeList,
+													Required: true,
+													Elem: &schema.Schema{
+														Type:         schema.TypeString,
+														ValidateFunc: validation.StringInSlice(quicksight.ColumnTagName_Values(), false),
+													},
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(1, 128),
+			},
+
+			"permissions": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MinItems: 1,
+				MaxItems: 64,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"actions": {
+							Type:     schema.TypeList,
+							Required: true,
+							MinItems: 1,
+							MaxItems: 16,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+						"principal": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringLenBetween(1, 256),
+						},
+					},
+				},
+			},
+
+			"physical_table_map": {
+				Type:     schema.TypeSet,
+				Required: true,
+				MaxItems: 32,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"physical_table_map_id": {
```

Comment:
```suggestion
						"physical_table_id": {
```
The ID here is referring to the unique key identifying each individual `PhysicalTable` -- so I think it makes more sense to remove "map" here.

Additionally, I believe that it is _this_ ID that is being referenced in `logical_table_map.0.source` as `physical_table_id` -- so it would be clean for those to match.